### PR TITLE
Support opt-in UTF-8 offsets

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1391,6 +1391,12 @@ pub struct ClientCapabilities {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub general: Option<GeneralClientCapabilities>,
 
+    /// Unofficial UT8-offsets extension.
+    ///
+    /// See https://clangd.llvm.org/extensions.html#utf-8-offsets.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub offset_encoding: Option<Vec<String>>,
```
Comment (kjeremy) [inline src/lib.rs:1398]: This should really be under `experimental` but I guess we don't control that.

**Comment by kjeremy** [comment]
Would it make sense to hide these under a feature flag?

**Comment by Marwes** [comment]
Probably under `proposed` I guess.

**Comment by matklad** [comment]
good call, added cfg!

**Pull request merged**